src/types/macros.ts

```diff
@@ -5,22 +5,25 @@ export interface MacroMap {
   [key: string]: string;
 }
 
+/**
+ * This function compares two MacroMap objects, and returns true if all properties
+ * are the same
+ * @param first first MacroMap object
+ * @param second second MacroMap object
+ * @returns boolean, true if they match else false
+ */
 export function macrosEqual(first: MacroMap, second: MacroMap): boolean {
-  // TODO why are the DIDs not matching?
   const { DID: firstDid, ...firstOthers } = first;
-  const { DID: secondDid, ...secondOthers } = first;
+  const { DID: secondDid, ...secondOthers } = second;
```

Comment:
Looks like you've caught another bug here. Is this one tested?

Reply:
It's tested now.